test/parallel/test-cli-eval.js

```diff
@@ -113,3 +113,16 @@ child.exec(nodejs + ` -e 'require("child_process").fork("${emptyFile}")'`,
   assert.strictEqual(proc.stderr, '');
   assert.strictEqual(proc.stdout, 'start\nbeforeExit\nexit\n');
 }
+
+[ '-arg1',
+  '-arg1 arg2 --arg3',
+  '--',
+].forEach(function(args) {
+  const evalopt = ' --eval "console.log(process.argv.slice(1).join(\' \'))"';
+  child.exec(nodejs + evalopt + ' -- ' + args,
```

Comment:
What I meant by my earlier comment about readability is that you can do this:

```js
const cmd = `${nodejs}${evalopt} -- ${args}`;

child.exec(cmd, common.mustCall(function(err, stdout, stderr) {
  ...
});
```

It gets rid of that awkward looking indented function.